tests/Unit/ControlSystem/Test_PiecewisePolynomial.cpp

```diff
@@ -0,0 +1,141 @@
+#include "tests/Unit/TestHelpers.hpp"
+
+SPECTRE_TEST_CASE("Unit.ControlSystem.FunctionsOfTime.PiecewisePolynomial",
+                  "[ControlSystem][Unit]") {
+  double t = 0.0;
+  const double dt = 0.6;
+  const double final_time = 4.0;
+  const constexpr size_t deriv_order = 3;
+
+  // test two component system (x**3 and x**2)
+  const std::array<DataVector, deriv_order + 1> init_func{
+      {{0.0, 0.0}, {0.0, 0.0}, {0.0, 2.0}, {6.0, 0.0}}};
+  FunctionsOfTime::PiecewisePolynomial<deriv_order> f_of_t(t, init_func);
+
+  while (t < final_time) {
+    const auto& lambdas = f_of_t(t);
+
+    CHECK(approx(lambdas[0][0]) == cube(t));
+    CHECK(approx(lambdas[0][1]) == square(t));
+
+    t += dt;
+    f_of_t.update(t, {6.0, 0.0});
+  }
+  // test time_bounds function
+  const auto& t_bounds = f_of_t.time_bounds();
+  CHECK(t_bounds[0] == 0.0);
+  CHECK(t_bounds[1] == 4.2);
+}
+
+SPECTRE_TEST_CASE(
+    "Unit.ControlSystem.FunctionsOfTime.PiecewisePolynomial.FirstDerivs",
+    "[ControlSystem][Unit]") {
+  double t = 0.0;
+  const double dt = 0.6;
+  const double final_time = 4.0;
+  const constexpr size_t deriv_order = 3;
+  const constexpr size_t ret_deriv_order = 1;
+
+  // test two component system (x**3 and x**2)
+  const std::array<DataVector, deriv_order + 1> init_func{
+      {{0.0, 0.0}, {0.0, 0.0}, {0.0, 2.0}, {6.0, 0.0}}};
+  FunctionsOfTime::PiecewisePolynomial<deriv_order> f_of_t(t, init_func);
+
+  while (t < final_time) {
+    const auto& lambdas = f_of_t.operator()<ret_deriv_order>(t);
+
+    CHECK(lambdas.size() == ret_deriv_order + 1);
+    CHECK(approx(lambdas[1][0]) == 3.0 * square(t));
+    CHECK(approx(lambdas[1][1]) == 2.0 * t);
+
+    t += dt;
+    f_of_t.update(t, {6.0, 0.0});
+  }
+}
+
+SPECTRE_TEST_CASE(
+   "Unit.ControlSystem.FunctionsOfTime.PiecewisePolynomial.NonConstDerivUpdate",
+   "[ControlSystem][Unit]") {
+  double t = 0.0;
+  const double dt = 0.6;
+  const double final_time = 4.0;
+  const constexpr size_t deriv_order = 2;
+
+  // initally x**2, but update with non-constant 2nd deriv
+  const std::array<DataVector, deriv_order + 1> init_func{
+      {{0.0}, {0.0}, {2.0}}};
+  FunctionsOfTime::PiecewisePolynomial<deriv_order> f_of_t(t, init_func);
+
+  while (t < final_time) {
+    t += dt;
+    f_of_t.update(t, {3.0 + t});
+  }
+  const auto& lambdas = f_of_t(t);
+  CHECK(approx(lambdas[0][0]) == 33.948);
+  CHECK(approx(lambdas[1][0]) == 19.56);
+  CHECK(approx(lambdas[2][0]) == 7.2);
+}
+
+SPECTRE_TEST_CASE(
+    "Unit.ControlSystem.FunctionsOfTime.PiecewisePolynomial.WithinRoundoff",
+    "[ControlSystem][Unit]") {
+  const constexpr size_t deriv_order = 3;
+  const size_t n_derivs = 3;
+  const std::array<DataVector, deriv_order + 1> init_func{
+      {{1.0, 1.0}, {3.0, 2.0}, {6.0, 2.0}, {6.0, 0.0}}};
+  FunctionsOfTime::PiecewisePolynomial<deriv_order> f_of_t(1.0, init_func);
+  f_of_t.update(2.0, {6.0, 0.0});
+  const auto& lambdas0 = f_of_t(1.0 - 5.0e-16);
+  CHECK(approx(lambdas0[0][0]) == 1.0);
```

Comment:
I think you can CHECK_ITERABLE_APPROX(lambdas0,  init_func); instead of the CHECK